### PR TITLE
add blacklist-regex and whitelist-regex support

### PR DESCRIPTION
**Pull request by taybin**
Finally implemented #21

### CODE REVIEW

File: main/main.go [hunk 1]
```diff
@@ -377,10 +377,13 @@ func replicateAction(c *cli.Context) error {
...
 	bl := c.GlobalStringSlice(config.VAR_NAME_BLACKLIST)
+	wlr := c.GlobalStringSlice(config.VAR_NAME_WHITELIST_REGEX)
+	blr := c.GlobalStringSlice(config.VAR_NAME_BLACKLIST_REGEX)
 	var tablelist []string
 	var whitelist bool
-	if len(wl) != 0 && len(bl) != 0 {
-		return errors.New("'whitelist' and 'blacklist' are mutually exclusive. Use one or the other but not both")
+	var regex bool
+	if len(wl) != 0 && len(bl) != 0 && len(wlr) != 0 && len(blr) != 0 {
+		return errors.New("'whitelist', 'whitelest-regex', 'blacklist', and 'blacklist-regex' are mutually exclusive. Use one or the other but not both")
```
Comment (stlava) [inline main/main.go]: Typo here `whitelest-regex`

Comment (taybin) [reply main/main.go]: Fixed.  Nice catch.

File: filter/filter.go [hunk 2]
```diff
@@ -151,9 +156,14 @@ func (f *Filter) Start() {
 		// check to see if relation (table name) is found in the lsit
 		found := false
 		for _, item := range f.tablelist {
-			if msg.Pr.Relation == item {
+			if f.regex == false && msg.Pr.Relation == item {
 				found = true
 				break
+			} else if f.regex == true {
+				if matched, _ := regexp.MatchString(item, msg.Pr.Relation); matched {
```
Comment (stlava) [inline filter/filter.go]: For performances please compile the set of regexes outside of the outer for loop.

Comment (taybin) [reply filter/filter.go]: Done.

**Pull request merged**